addons/road-generator/nodes/road_container.gd

```diff
@@ -317,6 +317,72 @@ func get_segments() -> Array:
 	return segs
 
 
+## Recursively gets all RoadContainers within a root node
+func get_all_road_containers(root: Node)->Array:
+	var nodes: Array = []
+	var dist: float
+
+	for n in root.get_children():
+		if n.get_child_count() > 0:
+			if n.has_method("is_road_container"):
+				nodes.append(n)
+			nodes.append_array(get_all_road_containers(n))
+		else:
+			if n.has_method("is_road_container"):
+				nodes.append(n)
+	return nodes
+
+
+## Snaps a RoadContainer's RoadPoint to a target RoadPoint
+## and moves the RoadContainer along with it.
+func snap_to_road_point(sel_rp: RoadPoint, tgt_rp: RoadPoint):
+	# Add a placeholder at selected RoadPoint position
+	var rc_par = get_parent()
+	var spa_sel_rp: Position3D = Position3D.new()
+	rc_par.add_child(spa_sel_rp)
+	spa_sel_rp.global_transform = sel_rp.global_transform
+
+	# Add a placeholder at RoadContainer position and
+	# make it a child of RoadPoint placeholder
+	var spa_sel_rc: Position3D = Position3D.new()
```

Comment:
This works for now, but I do feel could probably get around the need for this by using xform or xform_inv to get transforms in the right space, as opposed to adding and then clearing a node which is a more expensive operation.